apps/test/unit/code-studio/components/progress/LessonLockDialogTest.jsx

```diff
@@ -1,112 +1,188 @@
 import {LockStatus} from '@cdo/apps/code-studio/lessonLockRedux';
 
+// This * import allows us to stub out the SectionSelector component
+import * as SectionSelector from '@cdo/apps/code-studio/components/progress/SectionSelector';
+
+const fakeSectionId = 42;
+
 const MINIMUM_PROPS = {
-  isOpen: false,
+  isOpen: true,
   handleClose: () => {},
   initialLockStatus: [],
-  selectedSectionId: null,
+  selectedSectionId: fakeSectionId,
   saving: false,
   saveDialog: () => {}
 };
 
-const fakeSectionId = 1;
+// Helper function to get the list of rows in the student table.
+const getStudentRows = wrapper =>
+  wrapper.find('#ui-test-student-table tbody tr');
+
+// Helper function to verify that the radio buttons in the given student row
+// correctly reflect the expected lock status.
+const verifyRowLockStatus = (row, expectedLockStatus) => {
+  const radioButtons = row.find('input');
+  const lockedSelected = radioButtons.at(0).props().checked;
+  expect(lockedSelected).to.equal(expectedLockStatus === LockStatus.Locked);
+  const editableSelected = radioButtons.at(1).props().checked;
+  expect(editableSelected).to.equal(expectedLockStatus === LockStatus.Editable);
+  const readonlySelected = radioButtons.at(2).props().checked;
+  expect(readonlySelected).to.equal(
+    expectedLockStatus === LockStatus.ReadonlyAnswers
+  );
+};
+
+describe('LessonLockDialog with stubbed section selector', () => {
+  // Stub out <SectionSelector> so we can test without Redux.
+  let sectionSelectorStub;
+  beforeEach(() => {
+    sectionSelectorStub = sinon.stub(SectionSelector, 'default');
+    sectionSelectorStub.callsFake(props => null);
+  });
+  afterEach(() => {
+    sectionSelectorStub.restore();
+  });
+
+  it('renders with minimal props', () => {
+    const wrapper = mount(<LessonLockDialog {...MINIMUM_PROPS} />);
+    expect(wrapper).not.to.be.null;
+    expect(wrapper.text()).not.to.be.empty;
+  });
 
-describe('LessonLockDialog', () => {
-  it('renders with a selected section', () => {
-    const wrapper = shallow(
-      <LessonLockDialog {...MINIMUM_PROPS} selectedSectionId={fakeSectionId} />
+  it('renders no content when closed', () => {
+    const wrapper = mount(
+      <LessonLockDialog {...MINIMUM_PROPS} isOpen={false} />
     );
     expect(wrapper).not.to.be.null;
+    expect(wrapper.text()).to.be.empty;
+  });
+
+  it('renders student row with name and lock status', () => {
+    const wrapper = mount(
+      <LessonLockDialog
+        {...MINIMUM_PROPS}
+        initialLockStatus={[{name: 'fakeName', lockStatus: LockStatus.Locked}]}
+      />
+    );
+
+    expect(getStudentRows(wrapper)).to.have.length(1);
+    const studentRow = getStudentRows(wrapper).at(0);
+    expect(
+      studentRow
+        .find('td')
+        .at(0)
+        .text()
+    ).equals('fakeName');
+    verifyRowLockStatus(studentRow, LockStatus.Locked);
   });
 
   it('updates lock status from new props if not saving', () => {
```

Comment:
Is this what you were talking about at standup? This feels confusing to me- that if we are *not* saving something, we have to update the props

Reply:
This bit of the data-handling logic is confusing.  The short version is that when we are saving, we don't want to apply new lock status (that presumably is being applied because of a change to the lock status in redux).  My guess is that the intent here is to "freeze" the dialog when a save request is pending (kind of like when a submit button on a shopping site gets grayed out after you click on it).  I'm hoping to improve this a bit in the next few PRs.